### PR TITLE
ISPN-13453 IracExponentialBackOffTest random failures

### PR DESCRIPTION
**Pull request by pruivo**
Concurrency issue between ControlledExponentialBackOff asynBackOff() and
release()

- asyncBackOff() adds the event.BACK_OFF
- main thread is unblocked
- release() is invoked and creates a new CompletableFuture (CF)
- asyncBackOff() returns the new CF and IRAC gets stucks in there since
  the CF is never completed

https://issues.redhat.com/browse/ISPN-13453

I think both failures are related since I'm unable to reproduce them locally. 
:crossed_fingers:

### CODE REVIEW

File: core/src/test/java/org/infinispan/xsite/irac/IracExponentialBackOffTest.java [hunk 1]
```diff
@@ -109,7 +109,8 @@ public void testNoBackoffOnOtherException(Method method) throws InterruptedExcep
    }
 
    private void doTest(Method method, Supplier<Throwable> throwableSupplier) throws InterruptedException {
-      backOff.drainPermits();
+      backOff.release();
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/xsite/irac/IracExponentialBackOffTest.java:112]: Wouldn't these first 4 lines fit better in an `@AfterMethod` method?

**Comment by danberindei** [comment]
@pruivo I ran the core tests with `taskset -c 1,2` on your branch and I got lots of failures in `IracCustomConflictTest` and `Irac3SitesConflictTest`, with traces like this:

```
18:17:17,759 ERROR (testng-Test:[]) [TestSuiteProgress] Test failed: org.infinispan.xsite.irac.IracCustomConflictTest.testRemove[O_TX, P_TX]
java.lang.AssertionError: Updated keys map is not empty!
	at org.testng.AssertJUnit.fail(AssertJUnit.java:59) ~[testng-6.14.3.jar:?]
	at org.infinispan.test.AbstractInfinispanTest.eventually(AbstractInfinispanTest.java:237) ~[test-classes/:?]
	at org.infinispan.test.AbstractInfinispanTest.eventually(AbstractInfinispanTest.java:253) ~[test-classes/:?]
	at org.infinispan.test.AbstractInfinispanTest.eventually(AbstractInfinispanTest.java:390) ~[test-classes/:?]
	at org.infinispan.xsite.AbstractMultipleSitesTest.assertNoDataLeak(AbstractMultipleSitesTest.java:146) ~[test-classes/:?]
	at org.infinispan.xsite.irac.IracCustomConflictTest.doTest(IracCustomConflictTest.java:204) ~[test-classes/:?]
	at org.infinispan.xsite.irac.IracCustomConflictTest.testRemove(IracCustomConflictTest.java:82) ~[test-classes/:?]
```

I assume those are not related, but the fact that `IracExponentialBackOffTest` only checks `IracManager` is empty at the start of the test methods got me thinking for a second.

**Comment by pruivo** [comment]
@danberindei I'm aware. See https://issues.redhat.com/browse/ISPN-13442 :+1: 
I added the check before the test because, if the `updateKeys` map is not empty, the back-off state will be different from expected make the test fail. I would rather know the test failed because of the map instead of trying to investigate if there is something wrong with the back-off logic.

File: core/src/test/java/org/infinispan/xsite/irac/IracExponentialBackOffTest.java (hunk 1 above)
Comment (pruivo) [reply core/src/test/java/org/infinispan/xsite/irac/IracExponentialBackOffTest.java]: done

**Comment by danberindei** [comment]
Merged, thanks Pedro!

I forgot about that issue, and it was a bit scary seeing 12 test failures!

**Pull request merged**